app/code/Magento/Customer/view/frontend/web/js/show-password.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery'
+], function ($) {
+    'use strict';
+
+    $.widget('mage.showPassword', {
```

Comment:
Could the widget be replaced with UI component to follow the latest Magento development best practices?
https://devdocs.magento.com/guides/v2.4/ui_comp_guide/bk-ui_comps.html

Reply:
I will update the code based on the best practices. Thanks for the review.

Reply:
@sivaschenko Can you please provide me some examples of the existing UI component implementation? That would be helpful for me to refactor the code.

Shall I use the `<script type="text/x-magento-init">` to initialize the script for show/hide password functionality ?

Reply:
Yes, example: 

```
<div class="field choice" data-bind="scope: 'showPassword'">
    <!-- ko template: getTemplate() --><!-- /ko -->
</div>
<script type="text/x-magento-init">
        "*": {
            "Magento_Ui/js/core/app": {
                "components": {
                    "showPassword": {
                        "component": "Magento_Customer/js/show-password",
                        "passwordSelector": "#current-password,#password,#password-confirmation", //may be an array as well
                        "template": "the checkbox can be moved to a component template"
                    },
```

Reply:
@sivaschenko I have updated the code using ui component. Please review it.